packages/web-components/src/components/rux-pop-up-menu/rux-pop-up-menu.tsx

```diff
@@ -1,276 +1,300 @@
+//@ts-nocheck
 import {
+    Prop,
+    Watch,
+    Element,
     Component,
     Host,
     h,
-    Prop,
-    Element,
-    EventEmitter,
+    State,
     Event,
+    EventEmitter,
     Method,
-    Watch,
-    State,
     Listen,
+    Event,
 } from '@stencil/core'
+import {
+    Placement,
+    computePosition,
+    arrow,
+    offset,
+    flip,
+    autoUpdate,
+} from '@floating-ui/dom'
 
 /**
- * @slot menu-end - Area below the menu list to insert elements. For example, confirmation and/or cancel button group.
- * @part container - the container for the pop-up-menu
+ * @slot (default) - The contents for rux-pop-up-menu
+ * @slot trigger - The trigger element for rux-pop-up-menu
+ *
+ * @part container - the container of rux-pop-up-menu
+ * @part trigger-container - the container of the pop-up trigger
+ * @part popup-content - the content that is shown when rux-pop-up-menu is opened
+ * @part arrow - the arrow pointing to the trigger of rux-pop-up-menu
  */
-
 @Component({
     tag: 'rux-pop-up-menu',
     styleUrl: 'rux-pop-up-menu.scss',
     shadow: true,
 })
 export class RuxPopUpMenu {
-    @Element() el!: HTMLRuxPopUpMenuElement
-
-    @State() anchorBounds?: DOMRect
-    @State() menuBounds?: DOMRect
-
-    /**
-     * Optional element to trigger opening and closing of the menu.
-     * If none is supplied the element where aria-controls === menu id will be assigned
-     */
-    @Prop({ mutable: true }) triggerEl?: HTMLElement
+    private trigger!: HTMLElement
+    private content!: HTMLElement
+    private arrowEl?: HTMLElement
+    private _positionerCleanup: ReturnType<typeof autoUpdate> | undefined
 
-    /**
-     * Element to anchor the menu to. If none is given the menu will anchor
-     * to the trigger element where aria-controls === menu id
-     */
-    @Prop({ mutable: true }) anchorEl?: HTMLElement
-    @Watch('triggerEl')
-    @Watch('anchorEl')
-    tieElements() {
-        this._bindElements()
-    }
-
-    /**
-     * Boolean which controls when to show the menu
-     */
-    @Prop({ reflect: true, mutable: true }) open: boolean = false
-    @Watch('open')
-    openMenu() {
-        this._toggleOpenClose()
-    }
+    @Element() el!: HTMLRuxPopUpMenuElement
 
     /**
-     * Emitted when the menu is about to open.
+     * @prop open - determines if the pop up is open or closed
      */
-    @Event({ eventName: 'ruxmenuwillopen' })
-    ruxMenuWillOpen!: EventEmitter<void>
-
+    @Prop({ mutable: true }) open = false
     /**
-     * Emitted when the menu is about to close
+     * @prop placement - the placement of the pop up relative to it's slotted trigger element. A list of
+     * acceptable placements can be found at [floatingui.com/docs](https://floating-ui.com/docs/computePosition#placement)
```

Comment:
lets pretend like we're theranos and floating ui is our seimens machine. the end user shouldn't need to know anything about floating ui. it'll make it easier to rip it out and replace it with something cool like if we ever get a native <popover>